scripts/install/sonarrv3.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+# Sonarr v3 installer
+# Flying sauasges for swizzin 2020
+
+if [[ -f /tmp/.install.lock ]]; then
+  export log="/root/logs/install.log"
+else
+  log="/root/logs/swizzin.log"
+fi
+
+#shellcheck source=sources/functions/ask
+. /etc/swizzin/sources/functions/ask
+
+#Handles existing v2 instances
+_sonarrv2_flow(){
+    v2present=false
+    if [[ -f /install/.sonarr.lock ]]; then
+        v2present=true
+    fi
+    if dpkg -l | grep nzbdrone > /dev/null 2>&1 ; then
+        v2present=true
+    fi
+
+    if [[ $v2present == "true" ]]; then
+        echo "Sonarr v2 is detected."
+        echo "Continuing will migrate your current v2 installation." | tee -a $log
+        echo "You can read more about the migration at https://docs.swizzin.ltd/applications/sonarrv3#migrating-from-v2"
+        echo "An additional copy of the backup will be made into /root/sonarrv2.bak/" | tee -a $log
+        if ! ask "Do you want to continue?" N; then
+            exit 0
+        fi
+
+        if ask "Would you like to trigger a Sonarr-side backup?" Y; then
+            echo "Backing up Sonarr v2" | tee -a $log
+            if [[ -f /install/.nginx ]]; then
+                address="http://127.0.0.1:8989/sonarr/api"
+            else
+                address="http://127.0.0.1:8989/api"
+            fi
+
+            [[ -z $sonarrv2owner ]] && sonarrv2owner=$(cut -d: -f1 < /root/.master.info)
+            if [[ ! -d /home/"${sonarrv2owner}"/.config/NzbDrone ]]; then
+                echo "No Sonarr config folder found for $sonarrv2owner. Exiting" | tee -a $log
+                exit 1
```

Comment:
Is exiting the right thing to do here?

If sonarr is installed but the config cannot be found, should we simply just instruct users to uninstall v2 and exit or just ask them if we want to continue anyway? Otherwise we could give them an actionable item such as:

"No Sonarr config folder found for /home/"${sonarrv2owner}"/.config/NzbDrone. Does it exist?"

Perhaps a `read -p` could be used for the user to confirm that continuing may not have the results expected or that they should quit the script themselves now.

Reply:
They could just re-start the installation without the migration too, and we do tell them about the docs at the beginning so I suppose that's the first thing they'd check. I'd rather do that than give them a way to do a botched migration.